### PR TITLE
Issue #12168: failure of update github.io to clone checkstyle.github.io

### LINKED ISSUE #12168: failure of update github.io to clone checkstyle.github.io
https://github.com/checkstyle/checkstyle/runs/8179173614?check_suite_focus=true
failure of update github.io
```
Clone by ssh only to avoid passwords on push ...
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.114.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.113.3' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
Warning: Permanently added the ECDSA host key for IP address '140.82.114.4' to the list of known hosts.
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Cloning into 'checkstyle.github.io'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
./.ci/update-github-io.sh: line 23: cd: .ci-temp/checkstyle.github.io: No such file or directory
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #12168 

https://github.com/stoyanK7/checkstyle/actions/runs/3448827113

### CODE REVIEW

**Review by romani — APPROVED** [review]
Ok to merge.

It is interesting to see if secret.PAT is default variable provided by GitHub or not.
I thought it is not.

**Comment by romani** [comment]
```
[ERROR] [checkstyle] [ERROR] C:\projects\checkstyle\.github\workflows\
update_github_io.yml:41: 
Line should not be longer than 100 symbols [lineLength]

```

**Comment by stoyanK7** [comment]
@romani It is not, I set it up.

**Comment by romani** [comment]
@stoyanK7 , please fix violations.
You can prevent this by execution of mvn verify on your local to make sure you follow all (uncountable) rules even on non java files.

https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-checkstyle-report

**Comment by romani** [comment]
>I change to secrets.GITHUB_TOKEN

Please try. It might not work are it is cross repository activities, and not simple token is required

**Comment by stoyanK7** [comment]
Indeed, getting 403 using `GITHUB_TOKEN`.
https://github.com/stoyanK7/checkstyle/actions/runs/3452309394/jobs/5762121454#step:5:12246
```
Push site content to remote ...
We do force to avoid history changes, we do not need history as github.io shows only HEAD.
remote: Permission to stoyanK7/checkstyle.github.io.git denied to github-actions[bot].
fatal: unable to access 'https://github.com/stoyanK7/checkstyle.github.io/': The requested URL returned error: 403
Error: Process completed with exit code 128.
```

**Comment by romani** [comment]
Ok, make it will with PAT

**Comment by stoyanK7** [comment]
Strange why though. [`GITHUB_TOKEN` has all permissions possible](https://github.com/stoyanK7/checkstyle/actions/runs/3452358540/jobs/5762212398#step:1:17)
```
GITHUB_TOKEN Permissions
  Actions: write
  Checks: write
  Contents: write
  Deployments: write
  Discussions: write
  Issues: write
  Metadata: read
  Packages: write
  Pages: write
  PullRequests: write
  RepositoryProjects: write
  SecurityEvents: write
  Statuses: write
```

**Comment by romani** [comment]
But for current repository, you try to push to another repo

**Comment by stoyanK7** [comment]
You are right, I assumed it is universal for all repositories under one's name. From docs:
> The token's permissions are limited to the repository that contains your workflow.

Reverting back to `secrets.PAT` but then you have to create such a token for `checkstyle`, correct?

**Comment by stoyanK7** [comment]
Ok, ready
Repo: https://github.com/stoyanK7/checkstyle.github.io
Job: https://github.com/stoyanK7/checkstyle/actions/runs/3452495232

File: .ci/update-github-io.sh [hunk 1]
```diff
@@ -17,9 +17,3 @@ git checkout "checkstyle-$TARGET_VERSION"
 
 echo "Generating web site"
 mvn -e --no-transfer-progress site -Pno-validations -Dmaven.javadoc.skip=false
-
-echo "Clone by ssh only to avoid passwords on push ..."
```
Comment (romani) [inline .ci/update-github-io.sh:21]: This script becomes weird.
Name is not matching to what it does.
Let's have two shell scripts.
Generate-website.sh
Push-website-to-github-io.sh

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Review by romani — APPROVED** [review]

File: .github/workflows/update_github_io.yml [hunk 2]
```diff
@@ -18,15 +18,16 @@ jobs:
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          repository: stoyanK7/checkstyle
```
Comment (romani) [inline .github/workflows/update_github_io.yml:21]: Not good

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by stoyanK7** [comment]
Still testing. Will let you know when ready.

File: .github/workflows/update_github_io.yml (hunk 2 above)
Comment (stoyanK7) [reply .github/workflows/update_github_io.yml:21]: I know, running actions right now on local to make sure everything is good to go.

**Comment by stoyanK7** [comment]
Ok, ready
Job: https://github.com/stoyanK7/checkstyle/actions/runs/3460579127/jobs/5777259114
Repo: https://github.com/stoyanK7/checkstyle.github.io

File: .github/workflows/update_github_io.yml (hunk 2 above)
Comment (stoyanK7) [reply .github/workflows/update_github_io.yml]: Done. Should be good to go.

**Comment by romani** [comment]
PAT is created at https://github.com/settings/tokens
image

@stoyanK7 , is it enought ?

**Comment by stoyanK7** [comment]
@romani Yes, exactly. Creating and adding to project secrets should be enough.

**Comment by stoyanK7** [comment]
@romani ping

**Review by romani — APPROVED** [review]
ok to merge,
thanks for pinning.

**Pull request merged**